### ISSUE TITLE
plotting with column labels

### ISSUE THREAD

**Issue by geoHeil** [Microsoft/LightGBM#278]
Using the sklearn API I can fit a lightGBM booster easily. If the input is a pandas data frame the `feature_names` attribute is filled correctly (with the real names of the columns). It can be obtained via `clf._Booster.dump_model()['feature_names']`

But when plotting it like `lgb.plot_importance(clf, figsize=(14,15))`
![download](https://cloud.githubusercontent.com/assets/1694964/22508997/8dd8936c-e88c-11e6-96c8-f6698648f917.png)
These names are not chosen on the side. 

This could serve as an improvement for the plots, if a dependency on pandas and seaborn is ok.
```
gbmMetrics = pd.concat([pd.Series(featNames), pd.Series(importance)], axis=1)
lgbmMetrics.columns = ['featureName', 'importance']
metrics_sorted = lgbmMetrics.sort('importance', ascending=False)
g = sns.barplot(x="importance", y="featureName", data=metrics_sorted)
g.figure.set_size_inches(12, 9)
sns.plt.show()
```

**Comment by wxchan** [comment]
I plan to refine a better api for getting feature name, then work on this.

**Comment by wxchan** [comment]
@geoHeil you can try latest codes now. I think seaborn depends on matplotlib, does it have additional feature (matplotlib does not have)?

**Comment by geoHeil** [comment]
Yes nice looking plots out of the box. ;)
wxchan <notifications@github.com> schrieb am Fr. 3. Feb. 2017 um 17:03:
